xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml

```diff
@@ -942,8 +952,9 @@
   #set ($filterQuery = "#livetable_getFilterQuery(""${safe_tableAlias}.value"" $matchType false $valueCount $paramPrefix)")
   #set ($whereSql = "${whereSql} and ($filterQuery.trim())")
   #foreach ($filterValue in $filterValues)
-    #set ($actualMatchType = $request.getParameterValues("${colname}_match").get($foreach.count - 1))
-    #livetable_addFilterParam($filterValue $actualMatchType $whereParams "${paramPrefix}${foreach.count}")
+    #set ($matchType = $request.getParameterValues("${colname}_match").get($foreach.count - 1))
+    #livetable_initializeMatchType()
+    #livetable_addFilterParam($filterValue $matchType $whereParams "${paramPrefix}${foreach.count}")
```

Comment:
The match type used to add the parameter value is still not synchronized with the match type used to generated the constraint (where clause) for that parameter. In other words, ``${colname}_match`` seems to influence only the parameter value and not how the where clause is built.